### ISSUE TITLE
[BUG] reverse proxy

### ISSUE THREAD

**Issue by matmair** [inventree/InvenTree#3807]
I'm having some trouble getting the proxy auth working and I was wondering if I'm missing a setting somewhere. I'm using traefik + traefik-forward-auth for SSO, which passes the X-Forwarded-User header to an nginx container (for serving static files) running in front of inventree.

I have
```
remote_login_enabled: True
remote_login_header: X-Forwarded-User
```
in my `inventree.yml` file, and the nginx config was copied from [here](https://github.com/inventree/InvenTree/blob/master/docker/production/nginx.prod.conf). I've also tried the `INVENTREE_REMOTE_LOGIN=true` and `INVENTREE_REMOTE_LOGIN_HEADER=X-Forwarded-User`, but in both cases I keep getting redirected to the login screen.

Do I need to add to the `authentication_backends` and `middlewares` in the configuration yaml as well, or is there another configuration setting I'm missing?

_Originally posted by @larchen in https://github.com/inventree/InvenTree/issues/1693#issuecomment-1281897370_

**Comment by matmair** [comment]
@larchen I think it is better to handle this as a separate issue and not in a closed FR

**Comment by larchen** [comment]
Sorry about that — thanks for moving it here.

**Comment by matmair** [comment]
@larchen is the user already provisioned in your instance?

**Comment by larchen** [comment]
Yes, I've tried with an already provisioned user as well as a new user -- in both cases I just get redirected to the login screen.

Is there an easy way to log the request headers seen by the inventree server? I want to make sure that the header is at least getting set correctly.

**Comment by matmair** [comment]
I will have to look into this. What version are you running? Are you sure the headers get passed to InvenTree?

> Is there an easy way to log the request headers seen by the inventree server? I want to make sure that the header is at least getting set correctly.

You can do this by enabling the debug mode and opening a non-existent page - it will show the headers in the debug screen.

**Comment by matmair** [comment]
I can't reproduce on master with Nginx or caddy with forward auth.

@larchen What version are you running? How do you set the header and are you sure it is passed through?

**Comment by larchen** [comment]
I'm running the stable version of the docker image: `inventree/inventree:stable`. The header is set by traefik-forward-auth and it definitely gets set correctly for the other container services I'm running, but I'm not 100% sure the header is getting forwarded correctly by nginx to the inventree-server.

Sorry I've been a bit busy with some other work the past couple of days -- will update when I get a chance to poke around some more.

**Comment by matmair** [comment]
@larchen no rush; please report back once you have checked if the headers get passed (nginx can also log headers).

**Comment by larchen** [comment]
Was finally able to get the proxy auth working -- turns out the issue was that django modifies http header names, so the `X-Forwarded-User` header that the traefik middleware added was changed to `HTTP_X_FORWARDED_USER`. Everything was good once I changed my env variable to `INVENTREE_REMOTE_LOGIN_HEADER=HTTP_X_FORWARDED_USER`.

May be worth adding a note somewhere in the documentation for those not as familiar with django? A link to this [page](https://docs.djangoproject.com/en/4.1/ref/request-response/#django.http.HttpRequest.META) maybe?
